### ISSUE TITLE
Comparison operators

### ISSUE THREAD

**Issue by aldanor** [bluss/rust-ndarray#35]
Would it make sense to have `a.le(b)` (also `a.leq(b)`, `a.ge(b)`, `a.geq(b)`) return boolean arrays? (where `a, b : PartialEq` and properly broadcastable).

**Comment by bluss** [comment]
Yes, but the comparison operators can't return this. Slightly differently named methods may.

**Comment by aldanor** [comment]
I meant `PartialOrd` of course -- and forgot that it already uses these method names :)

So `ne`, `eq`, `ge`, `gt`, `le`, `lt` are taken by traits, I guess the names could be something like `equal`, `not_equal`, `less`, `less_or_equal` (or `less_than`, `leq_than`), or something like that..

**Comment by bluss** [comment]
It sounds like we'd want array masking then as well

**Comment by aldanor** [comment]
Yep, that's what it logically leads to :smile: 

But also enables implementing things like `min()`, `max()`, `ptp()` concisely.

**Comment by bluss** [comment]
We'll see that we can implement. My main focus now is just to improve contiguous array handling and f order arrays.

**Comment by nilgoyette** [comment]
Might be a good idea. In the meantime, one can use `map_collect`

    let lt = Zip::from(&a).and(&b).map_collect(|&a, &b| a < b);
